### DISCUSSION TITLE
Is there a best practice to invoke Cross-Channel Transaction ?

### DISCUSSION THREAD

**Discussion by Zrealshadow** [hyperledger/fabric#2571] [Q&A]
For example.
account A in  channel C1
account B in channel  C2
now we sent amount from account A to account B.
accountA.amount -= amount
accountB.amount += amount

How can we implement above scenario in Fabric

**Comment by yacovm** [comment]
You can't do it atomically, you need 2 transactions because each transaction reaches each channel in its own time.

**Reply by Zrealshadow** [reply]
Thanks for your reply.

I want to use channels to simulate the sharding technology in database.
You know, cross-sharding transaction in database is common. (For example, Account A in shard a, Account B in shard b, sentAmount from Account A to Account B).
Now I want to implement this chaincode and smart contract.

I have found many references, but no example code or tutorials are shared.

Would you mind giving me some hints or suggestion for this problem
